index.js

```diff
@@ -1,7 +1,9 @@
+let resolvedPromise
+
 module.exports = typeof queueMicrotask === 'function'
   ? queueMicrotask
-  : typeof Promise === 'function'
-    ? cb => Promise.resolve()
+  : (typeof Promise === 'function' ? (resolvedPromise = Promise.resolve()) : false)
```

Comment:
Wait a minute. This actually allocates a new promise each time... unless I'm missing something. I'll fix this up in another commit